lib/identity/client.js

```diff
@@ -766,6 +766,51 @@ class Client extends PartialClient {
     )
     return await validateResponseAsJSON(response)
   }
+  /**
+   * removeSingleSecret
+   * @param {String} secretID
+   * @param {String} version
+   * @param {String} secretName
+   * @param {String} secretType
+   * @return {Boolean} returns true if the secret was successfully removed
+   */
+  async removeSingleSecret(secretID, version, secretName, secretType) {
```

Comment:
From a usability perspective I would pass in a secret and then get those values in this method. instead of having 4 parameters that all come from the same object

Reply:
+1

Reply:
will do! i got so excited that it was finally working that I completely forgot we talked about this!

Reply:
maybe even just the secret meta.

Reply:
ah ah, don't cross abstractions if it can be avoided, even if its the easiest path forward at the moment. Conceptually secrets exists on top of records (hence why we refer to secretID and not recordID), our "public" sdk methods to work with secrets should accept secret objects, and pull out the lower level bits as needed to accomplish their tasks. If this was an e3db-clients-go lower level library sort of thing sure, but this is an SDK.